File: jdk/leetcode/RomanToInteger.java
package jdk.leetcode;

import java.util.HashMap;
import java.util.Map;

/***
 * 13. Roman to Integer
 */
public class RomanToInteger {
    public static void main(String[] args) {
        String s = "III";
        System.out.println(romanToInt(s));
    }

    public static int romanToInt(String s) {
        int result = 0;
        Map<Character,Integer> m = new HashMap<>();
        m.put('I', 1);
        m.put('V', 5);
        m.put('X', 10);
        m.put('L', 50);
        m.put('C', 100);
        m.put('D', 500);
        m.put('M', 1000);

        for ( int i = 0; i < s.length(); i ++) {
            if (i < s.length() - 1 && m.get(s.charAt(i)) < m.get(s.charAt(i + 1)))
                result -= m.getOrDefault(s.charAt(i), 0);
            else
                result += m.getOrDefault(s.charAt(i),0);
        }

        return result;
    }
}

File: python/leetcode/RomanToInteger.py
# 13. Roman to Integer
class RomanToInteger:
    def romanToInt(self, s: str) -> int:
        map = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        num = 0
        for i in range(len(s)):
            if i < len(s) - 1 and map[s[i]] < map[s[i + 1]]:
                num -= map.get(s[i])
            else:
                num += map.get(s[i])

        return num

# Call the twoSum method
romanToInteger = RomanToInteger()
result = romanToInteger.romanToInt("III")
print(result)